ddl/sequence.go

```diff
@@ -0,0 +1,178 @@
+		return errors.Trace(err)
+	}
+	return t.CreateSequenceAndSetSeqValue(schemaID, tbInfo, tbInfo.Sequence.Start)
+}
+
+func handleSequenceOptions(SeqOptions []*ast.SequenceOption, sequenceInfo *model.SequenceInfo) {
+	var (
+		minSetFlag   bool
+		maxSetFlag   bool
+		startSetFlag bool
+	)
+	for _, op := range SeqOptions {
+		switch op.Tp {
+		case ast.SequenceOptionIncrementBy:
+			sequenceInfo.Increment = op.IntValue
+		case ast.SequenceStartWith:
+			sequenceInfo.Start = op.IntValue
+			startSetFlag = true
+		case ast.SequenceMinValue:
+			sequenceInfo.MinValue = op.IntValue
+			minSetFlag = true
+		case ast.SequenceMaxValue:
+			sequenceInfo.MaxValue = op.IntValue
+			maxSetFlag = true
+		case ast.SequenceCache:
+			sequenceInfo.CacheValue = op.IntValue
+		case ast.SequenceNoCache:
+			sequenceInfo.Cache = false
+		case ast.SequenceCycle:
+			sequenceInfo.Cycle = true
+		case ast.SequenceNoCycle:
+			sequenceInfo.Cycle = false
+			// TODO: TiDB haven't actually support ORDER option by now.
+		}
+	}
+	// Fill the default value, min/max/start should be adjusted with increment's positive and negative.
+	if !(minSetFlag && maxSetFlag && startSetFlag) {
+		if sequenceInfo.Increment >= 0 {
+			if !minSetFlag {
+				sequenceInfo.MinValue = model.DefaultPositiveSequenceMinValue
+			}
+			if !startSetFlag {
+				sequenceInfo.Start = mathutil.MaxInt64(sequenceInfo.MinValue, model.DefaultPositiveSequenceStartValue)
+			}
+			if !maxSetFlag {
+				sequenceInfo.MaxValue = model.DefaultPositiveSequenceMaxValue
+			}
+		} else {
+			if !maxSetFlag {
+				sequenceInfo.MaxValue = model.DefaultNegativeSequenceMaxValue
+			}
+			if !startSetFlag {
+				sequenceInfo.Start = mathutil.MinInt64(sequenceInfo.MaxValue, model.DefaultNegativeSequenceStartValue)
+			}
+			if !minSetFlag {
+				sequenceInfo.MaxValue = model.DefaultNegativeSequenceMinValue
+			}
+		}
+	}
+}
+
+func validSequenceOptions(seqInfo *model.SequenceInfo) bool {
+	// To ensure that cache * increment will never overflows.
+	var maxIncrement int64
+	if seqInfo.Increment != 0 {
+		maxIncrement = math2.Abs(seqInfo.Increment)
+	} else {
+		maxIncrement = math.MaxInt16
+	}
+	if seqInfo.MaxValue >= seqInfo.Start &&
+		seqInfo.MaxValue > seqInfo.MinValue &&
+		seqInfo.Start >= seqInfo.MinValue &&
+		seqInfo.MaxValue != math.MaxInt64 &&
+		seqInfo.MinValue != math.MinInt64 &&
+		seqInfo.CacheValue < (math.MaxInt64-maxIncrement)/maxIncrement {
+		return true
+	}
+	return false
+}
+
+func buildSequenceInfo(stmt *ast.CreateSequenceStmt, ident ast.Ident) (*model.SequenceInfo, error) {
+	sequenceInfo := &model.SequenceInfo{
+		Cache:      model.DefaultSequenceCacheBool,
+		Cycle:      model.DefaultSequenceCycleBool,
+		Order:      model.DefaultSequenceOrderBool,
+		CacheValue: model.DefaultSequenceCacheValue,
+		Increment:  model.DefaultSequenceIncrementValue,
+	}
+
+	// Handle table comment options.
+	for _, op := range stmt.TblOptions {
+		switch op.Tp {
+		case ast.TableOptionComment:
```

Comment:
I can't sure if we need to support Comment.

Reply:
If we really plan to support comment, we have better use single `Comment` instead of `TableOption`

Reply:
This has been discussed in parser, actually its a option list. Take `comment xxx comment xxx` into consideration, only the last comment counts like collate and charset does.